dask/config.py

```diff
@@ -0,0 +1,304 @@
+from __future__ import print_function, division, absolute_import
+
+import ast
+import copy
+import os
+import sys
+
+
+no_default = '__no_default__'
+
+
+config_paths = [
+    '/etc/dask',
+    os.path.join(sys.prefix, 'etc', 'dask'),
+    os.path.join(os.path.expanduser('~'), '.config', 'dask'),
+    os.path.join(os.path.expanduser('~'), '.dask')
```

Comment:
Should `~/.dask` not be before `~/.config/dask`? IIUC the former is deprecated so I would expect that it does not win against the latter.

Reply:
> It doesn't do much concrete harm, except by spreading around the places to look for config.

I guess the use case I had in mind was:
* you have an old config in `~/.dask` that you forgot even existed
* you are trying to configure something in `~/.config/dask` because you have heard that this is the modern way of doing things. This does not have any effect because your config is overriden by the one in `~/.dask`

Maybe it is unlikely to happen because people do not tweak dask configuration files very often.

Reply:
Can we have a "list configurations" function? It should give the locations searched, and if anything was found in each, with timestamps.